utility/src/main/java/org/oppia/util/parser/UrlImageParser.kt

```diff
@@ -0,0 +1,81 @@
+package org.oppia.util.parser
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.Rect
+import android.graphics.drawable.BitmapDrawable
+import android.graphics.drawable.Drawable
+import android.text.Html
+import android.widget.TextView
+import com.bumptech.glide.request.target.SimpleTarget
+import com.bumptech.glide.request.transition.Transition
+import org.oppia.util.R
+import javax.inject.Inject
+import javax.inject.Singleton
+
+// TODO (#169) : Replace this with exploration asset downloader
+/** UrlImage Parser for android TextView to load Html Image tag. */
+@Singleton
+class UrlImageParser @Inject constructor(
+  @DefaultGcsPrefix private val gcsPrefix: String,
+  @DefaultGCSResource private val gcsResource: String,
+  @ImageDownloadUrlTemplate private var imageDownloadUrlTemplate: String
+) : Html.ImageGetter {
+
+  lateinit var htmlContentTextView: TextView
+  lateinit var context: Context
+  lateinit var entityType: String
+  lateinit var entityId: String
+
+  fun setImageData(htmlContentTextView: TextView, context: Context, entityType: String, entityId: String) {
```

Comment:
Suggest using a factory to set this via constructor (but still keep the necessary injections) per what I suggested above for the HtmlParser class. That's less fragile than relying on an external initialization method that might be forgotten to be called which would lead to unexpected results.

Reply:
Done as per suggestion, introduced factory class.